ProjFS.Mac/PrjFSKext/ProviderMessaging.cpp

```diff
@@ -124,6 +124,7 @@ void ProviderMessaging_AbortOutstandingEventsForProvider(VirtualizationRootHandl
     Mutex_Release(s_outstandingMessagesMutex);
 }
 
+#ifndef KEXT_UNIT_TESTING
```

Comment:
Thoughts on this approach?  I don't believe we have anything similar yet, and I can see this being the only function we'd want to leave out for now.  

There is also the swap approach suggested in this PR.  https://github.com/Microsoft/VFSForGit/pull/732#pullrequestreview-199062756

Reply:
I think the swap approach is going to be the long term solution, but for now this is fine, at least as long as we'r not trying to write tests for `ProviderMessaging_TrySendRequestAndWaitForResponse` itself. We can eventually also expand MockCalls to generate a function to swap with automatically, so we don't have to have these dummy functions forwarding to `MockCalls::RecordFunctionCall` everywhere; it'll be super slick to use if we can pull off the gnarly C++ template back-end.

Reply:
> Thoughts on this approach?

@jeschu1 I don't think we need the `#ifndef KEXT_UNIT_TESTING` here, because it looks like we don't need to compile `ProviderMessaging.cpp` as part of `PrjFSKextTestable` at all (I pushed a branch with this change, see https://github.com/wilbaker/VFSForGit/commit/6acd456c9a60438a61a5b9961652565bb50da922)

I think this approach is OK for now if we find spots that need it, but at this point I'm not sure if the unit tests need to call any of the functions in `ProviderMessaging.cpp`.

cc: @pmj for his thoughts as well

Reply:
Related, can anyone remember if there was a need to compile `ProviderMessaging.cpp` as part of `Testable`?

I'm thinking this might have just been picked up by mistake (and didn't cause any issues until now).

Reply:
> Related, can anyone remember if there was a need to compile `ProviderMessaging.cpp` as part of `Testable`?
> 
> I'm thinking this might have just been picked up by mistake (and didn't cause any issues until now).

I think I included it when moving those functions to the new file to ensure that they successfully compiled in the user space/testable build environment, and that I hadn’t for example missed some headers. We can certainly take it back out of the testable build for now, I don’t think any of my WIP PRs will be testing those functions just yet.